TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/IDataverseAdapter.cs

```diff
@@ -105,4 +105,6 @@ public interface IDataverseAdapter
     Task ClearTeacherIdentityInfo(Guid identityUserId, DateTime updateTimeUtc);
 
     Task<bool> DoesTeacherHavePendingPIIChanges(Guid teacherId);
+
+    Task<List<dfeta_teacherstatus>> GetAllTeacherStatuses();
```

Comment:
Can we use `ReferenceDataCache` or a new query instead of adding to `IDataverseAdapter`?